src/tunnel.c

```diff
@@ -851,34 +852,39 @@ int ssl_connect(struct tunnel *tunnel)
 	}
 
 	/* Modify default TLS security levels */
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+	if (tunnel->config->min_tls <= 0)
+		tunnel->config->min_tls = TLS1_2_VERSION;
```

Comment:
I would set `tunnel->config->min_tls` unconditionally if left to <= 0, so you don't need to check for <=0 below. This actually might belong in default config, in main.c?
And at least change the default in a separate commit for history (this might deserve something in CHANGELOG.md...)